File: app/src/main/java/com/example/plantea/dominio/Categoria.java
package com.example.plantea.dominio;

import android.app.Activity;

import java.util.ArrayList;

public class Categoria {

    private int categoria;
    ArrayList<String> listaCategorias;
    GestionCategorias gestorCategorias = new GestionCategorias();

    public Categoria(){

    }

    public Categoria(int categoria) {
        this.categoria = categoria;
    }

    public int getCategoria() {
        return categoria;
    }

    public void setCategoria(int categoria) {
        this.categoria = categoria;
    }

    public ArrayList consultarCategorias(Activity actividad) {
        listaCategorias = new ArrayList<>();
        listaCategorias = gestorCategorias.listarCategorias(actividad);
        return listaCategorias;
    }

    public int obtenerCategoria (Activity actividad, String nombre){
        categoria = gestorCategorias.obtenerIdCategoria(actividad, nombre);
        return categoria;
    }

    public void crearCategoria (Activity actividad, String nombre){
        gestorCategorias.insertarSubcategoria(actividad, nombre);
    }

}

File: app/src/main/java/com/example/plantea/dominio/Categoria.kt
package com.example.plantea.dominio

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap

class Categoria {
    private var categoria = 0
    private var titulo: String? = null
    private var imagen: Bitmap? = null
    private var color: String? = null

    private var gestorCategorias = GestionCategorias()

    //getters
    fun getCategoria() = categoria
    fun getTitulo() = titulo
    fun getImagen() = imagen
    fun getColor() = color

    constructor()
    constructor(categoria: Int = 0, titulo: String?, imagen: Bitmap?, color: String?) {
        this.categoria = categoria
        this.titulo = titulo
        this.imagen = imagen
        this.color = color
    }

    fun obtenerCategoria(context: Context, nombre: String?, language: String): Int {
        return gestorCategorias.obtenerIdCategoria(context, nombre, language)
    }

    fun duplicateCategoria(context: Context, idUsuario: String, idCategoria: Int): Int {
        return gestorCategorias.duplicateCategoria(context, idUsuario, idCategoria)
    }

    fun obtenerCategoriasPrincipales(actividad: Activity?, idUsuario: String, language: String): ArrayList<Categoria> {
        return gestorCategorias.obtenerCategoriasPrincipales(actividad, idUsuario, language)
    }

    fun crearCategoria(actividad: Activity?, nombre: String?, imagen: ByteArray?, color: String, idUsuario: String): Int {
        return gestorCategorias.insertarCategoria(actividad, nombre, imagen, color, idUsuario)
    }

    fun eliminarCategoria(actividad: Activity?, idUsuario: String, idCategoria: Int) {
        gestorCategorias.eliminarCategoria(actividad, idUsuario, idCategoria)
    }

    fun checkCategoriaExiste(context: Context?, toString: String, idUsuario: String, language: String): Boolean {
        return gestorCategorias.checkCategoriaExiste(context, toString, idUsuario, language)
    }

}